QModManager/Utility/IOUtilities.cs

```diff
@@ -0,0 +1,180 @@
+            try
+            {
+                DirectoryInfo dirInfo = new DirectoryInfo(directory);
+                string toWrite = $"{GenerateSpaces(spaces)}|---+ {dirInfo.Name} ({ParseSize(CalculateFolderSizeRecursively(directory))})\n";
+
+                if (BannedFolders.Contains(dirInfo.Name) || BannedFolders.Contains($"{dirInfo.Parent.Name}/{dirInfo.Name}"))
+                {
+                    toWrite += $"{GenerateSpaces(spaces + 4)}`---- ({GetFileCountRecursively(directory)} elements not shown...)\n";
+                    return toWrite;
+                }
+
+                foreach (string dir in Directory.GetDirectories(directory))
+                {
+                    toWrite += GetFolderStructureRecursively(dir, spaces + 4);
+                }
+
+                string[] files = Directory.GetFiles(directory);
+                for (int i = 1; i <= files.Length; i++)
+                {
+                    FileInfo fileinfo = new FileInfo(files[i - 1]);
+                    if (i != files.Length)
+                        toWrite += $"{GenerateSpaces(spaces + 4)}|---- {fileinfo.Name} ({ParseSize(fileinfo.Length)})\n";
+                    else
+                        toWrite += $"{GenerateSpaces(spaces + 4)}`---- {fileinfo.Name} ({ParseSize(fileinfo.Length)})\n";
+                }
+
+                return toWrite;
+            }
+            catch (Exception e)
+            {
+                throw e;
+            }
+        }
+
+        private static string ParseSize(long lsize)
+        {
+            float size = lsize;
+            string unit;
+
+            if (size < 1024)
+            {
+                unit = "B";
+            }
+            else
+            {
+                size /= 1024;
+                if (size < 1024)
+                {
+                    unit = "KB";
+                }
+                else
+                {
+                    size /= 1024;
+                    if (size < 1024)
+                    {
+                        unit = "MB";
+                    }
+                    else
+                    {
+                        size /= 1024;
+                        if (size < 1024)
+                        {
+                            unit = "GB";
+                        }
+                        else
+                        {
+                            size /= 1024;
+                            unit = "TB";
+                        }
+                    }
+                }
+            }
+
+            string number = size.ToTwoDecimalString();
+            number.TrimEnd('0');
+            number.TrimEnd('.');
+
+            return number + unit;
+        }
+        private static long CalculateFolderSizeRecursively(string directory)
+        {
+            try
+            {
+                long size = 0;
+                foreach (string dir in Directory.GetDirectories(directory))
+                {
+                    size += CalculateFolderSizeRecursively(dir);
```

Comment:
The total size of folders seems irrelevant for trouble-shooting.  
Why not just show only the size of files and simply this?

Reply:
Addressed in commit c2d45a3